rustls/src/server/server_conn.rs

```diff
@@ -564,9 +564,6 @@ mod connection {
         /// Make a new ServerConnection.  `config` controls how
         /// we behave in the TLS protocol.
         pub fn new(config: Arc<ServerConfig>) -> Result<Self, Error> {
-            let mut common = CommonState::new(Side::Server);
-            common.set_max_fragment_size(config.max_fragment_size)?;
```

Comment:
Do we understand why lints didn't fire against this?

Reply:
It seems like `unused_assignments` only works for locals, not slots in arrays, tuples or structs. https://github.com/rust-lang/rust/issues/65467#issuecomment-994312394

I would have expected the assignment to `enable_secret_extraction` on a value that is immediately dropped to warn at least. The `set_max_fragment_size` line is OK as it potentially alters control flow.